commit d1aad9ec2b37560ddf2b2a362e30882adce8fb94
Author: Joo Aun Saw <jasaw@dius.com.au>
Date:   Mon Aug 19 16:36:30 2019 +1000

    fix h264_v4l2m2m encoder draining error
    
    When h264_v4l2m2m encoder is drained, it sets packet pts and size to
    zero rather than returning AVERROR_EOF.

diff --git a/ffmpeg.c b/ffmpeg.c
index beb87b1..a3a8701 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -968,11 +968,16 @@ static int ffmpeg_flush_codec(struct ffmpeg *ffmpeg){
                     my_packet_unref(ffmpeg->pkt);
                     return -1;
                 }
-                retcd = av_write_frame(ffmpeg->oc, &ffmpeg->pkt);
-                if (retcd < 0) {
-                    MOTION_LOG(ERR, TYPE_ENCODER, NO_ERRNO
-                        ,_("Error writing draining video frame"));
-                    return -1;
+                // v4l2_m2m encoder uses pts 0 and size 0 to indicate AVERROR_EOF
+                if ((ffmpeg->pkt.pts > 0) && (ffmpeg->pkt.size > 0)) {
+                    retcd = av_write_frame(ffmpeg->oc, &ffmpeg->pkt);
+                    if (retcd < 0) {
+                        MOTION_LOG(ERR, TYPE_ENCODER, NO_ERRNO
+                            ,_("Error writing draining video frame"));
+                        return -1;
+                    }
+                } else {
+                    recv_cd = AVERROR_EOF;
                 }
             }
             my_packet_unref(ffmpeg->pkt);
